tests/queries/0_stateless/01494_storage_join_persistency.sql

```diff
@@ -0,0 +1,8 @@
+DROP TABLE IF EXISTS join;
+
+CREATE TABLE join (k UInt64, s String) ENGINE = Join(ANY, LEFT, k) SETTINGS disable_persistency=1;
+
+DETACH TABLE join;
```

Comment:
What did I mean... we can insert some data,
validate that it will be present after DETACH/ATTACH if disable_persistency = 0
and validate that table will be empty after DETACH/ATTACH otherwise.

PS. Maybe better to name a setting `persistent` with default value 1?